File: 50-PowxN/50-PowxN.java
// Last updated: 7/31/2026, 8:29:11 PM
class Solution {
    public double Pow(double x, long N) {
        if (N==0) return 1; 
        double smol = Pow(x, N/2); 
        if (N%2==0) return smol*smol; 
        else return x*smol*smol;
    }
    public double myPow(double x, int n) {
        long N = n;
        if (N<0) {
            x = 1/x;
            N = -N;
        }
        return Pow(x,N);
    }
}

File: 50-PowxN/50-PowxN.py
# Last updated: 7/31/2026, 8:29:03 PM
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0 :
            return 1
        if n < 0 :
            x = 1/x
            n = -n
        smol = self.myPow(x,n//2)
        if n%2==0 : 
            return smol*smol
        else : 
            return x*smol*smol